### ISSUE TITLE
`VisibleDeprecationWarning` when using ragged parameter lists in tape mode

### ISSUE THREAD

**Issue by thisac** [PennyLaneAI/pennylane#1006]
#### Issue description
Using NumPy v.19.0, `dtype=object` must be specified when creating a ragged array. This happens e.g. in the [Doubly stochastic gradient descent demo](https://pennylane.ai/qml/demos/tutorial_doubly_stochastic.html) when running it with tape mode enabled, due to the `params` list containing ragged sequences, and then being cast into a NumPy array _without_ specifiying `dtype=object`.

```
pennylane/tape/tapes/jacobian_tape.py:499: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  params = np.array(params)
```

* *Expected behavior:* (What you expect to happen)
Running the demo (and other similar code) in tape mode succeeds without a `VisibleDeprecationWarning`.

* *Actual behavior:* (What actually happens)
A `VisibleDeprecationWarning` is raised whenever a ragged list of parameters are used with tape mode.

* *Reproduces how often:* (What percentage of the time does it reproduce?)
Every time!

* *System information:* (post the output of `import pennylane as qml; qml.about()`)
```
Name: PennyLane
Version: 0.14.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/theodor/Xanadu/pennylane
Requires: numpy, scipy, networkx, autograd, toml, appdirs, semantic-version
Required-by: PennyLane-Qchem, PennyLane-Lightning, PennyLane-Forest, PennyLane-PQ, PennyLane-SF, pennylane-qulacs, PennyLane-Cirq, PennyLane-qiskit
Platform info:           Linux-5.8.18-1-MANJARO-x86_64-with-arch-Manjaro-Linux
Python version:          3.7.6
Numpy version:           1.19.5
Scipy version:           1.4.1
Installed devices:
- lightning.qubit (PennyLane-Lightning-0.11.0)
- forest.numpy_wavefunction (PennyLane-Forest-0.9.0)
- forest.qpu (PennyLane-Forest-0.9.0)
- forest.qvm (PennyLane-Forest-0.9.0)
- forest.wavefunction (PennyLane-Forest-0.9.0)
- projectq.classical (PennyLane-PQ-0.9.0.dev0)
- projectq.ibm (PennyLane-PQ-0.9.0.dev0)
- projectq.simulator (PennyLane-PQ-0.9.0.dev0)
- strawberryfields.fock (PennyLane-SF-0.12.0.dev0)
- strawberryfields.gaussian (PennyLane-SF-0.12.0.dev0)
- strawberryfields.gbs (PennyLane-SF-0.12.0.dev0)
- strawberryfields.remote (PennyLane-SF-0.12.0.dev0)
- strawberryfields.tf (PennyLane-SF-0.12.0.dev0)
- qulacs.simulator (pennylane-qulacs-0.12.0.dev0)
- cirq.mixedsimulator (PennyLane-Cirq-0.13.0.dev0)
- cirq.pasqal (PennyLane-Cirq-0.13.0.dev0)
- cirq.qsim (PennyLane-Cirq-0.13.0.dev0)
- cirq.qsimh (PennyLane-Cirq-0.13.0.dev0)
- cirq.simulator (PennyLane-Cirq-0.13.0.dev0)
- qiskit.aer (PennyLane-qiskit-0.13.0.dev0)
- qiskit.basicaer (PennyLane-qiskit-0.13.0.dev0)
- qiskit.ibmq (PennyLane-qiskit-0.13.0.dev0)
- default.gaussian (PennyLane-0.14.0.dev0)
- default.mixed (PennyLane-0.14.0.dev0)
- default.qubit (PennyLane-0.14.0.dev0)
- default.qubit.autograd (PennyLane-0.14.0.dev0)
- default.qubit.jax (PennyLane-0.14.0.dev0)
- default.qubit.tf (PennyLane-0.14.0.dev0)
- default.tensor (PennyLane-0.14.0.dev0)
- default.tensor.tf (PennyLane-0.14.0.dev0)
```

**Comment by josh146** [comment]
@thisac I'm not sure about other demos, but for the Doubly Stochastic demo in particular, the reason for the warnings is that matrices are being used in the circuit without being marked as non-differentiable. As a result, the parameter-shift rule is being attempted on these parameters (lengthening the run time).

This can be fixed by adding `requires_grad=False` to the following lines:

```python
L143: H = np.array([[8, 4, 0, -6], [4, 0, 4, 0], [0, 4, 8, 0], [-6, 0, 0, 0]], requires_grad=False)
L300: A = np.sum(terms[idx], axis=0, requires_grad=False)
```

After making these two line changes, the visible deprecation warning also vanishes.

**Comment by thisac** [comment]
Thanks @josh146!
